Tests/OptimizelyTests-Common/OptimizelyUserContextTests_ForcedDecisions.swift

```diff
@@ -0,0 +1,392 @@
+//
+// Copyright 2021, Optimizely, Inc. and contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import XCTest
+
+class OptimizelyUserContextTests_ForcedDecisions: XCTestCase {
+
+    let kUserId = "tester"
+    
+    var optimizely: OptimizelyClient!
+    var eventDispatcher = MockEventDispatcher()
+    var decisionService: DefaultDecisionService!
+    var ups: OPTUserProfileService!
+
+    override func setUp() {
+        super.setUp()
+        
+        let datafile = OTUtils.loadJSONDatafile("decide_datafile")!
+        optimizely = OptimizelyClient(sdkKey: OTUtils.randomSdkKey,
+                                      eventDispatcher: eventDispatcher,
+                                      userProfileService: OTUtils.createClearUserProfileService())
+        decisionService = (optimizely.decisionService as! DefaultDecisionService)
+        ups = decisionService.userProfileService
+        try! optimizely.start(datafile: datafile)
+    }
+   
+    func testSetForcedDecision_flagToDecision() {
+        let featureKey = "feature_1"
+        let user = optimizely.createUserContext(userId: kUserId)
+        _ = user.setForcedDecision(flagKey: featureKey,
```

Comment:
Can we also confirm the expected result of this call?

Reply:
A good catch. I added a separate test to validate return status for all calls.

Reply:
We have test cases for invalid variation keys - "testSetForcedDecision_invalid()".
Other cases with invalid flagKey and ruleKey will be considered as mismatches, so no explicit testing.